### DISCUSSION TITLE
How can I upload a file to storage using cURL?

### DISCUSSION THREAD

**Discussion by michrome** [supabase/supabase#1027] [Questions] [answered]
Hey all

Assuming my Buckets and Objects have Row Level Security disabled (dangerous, I know), how would I upload a file to storage using curl? I tried the following:

```bash
## cURL
curl -X "POST" "https://locgxsdxxxxxxxxxx.supabase.co/storage/v1/object/bucket1/my-file.jpg" \
     -H 'Accept: */*' \
     -H 'Origin: https://app.supabase.io' \
     -H 'Host: locgxsdxxxxxxxxxx.supabase.co' \
     -H 'Accept-Encoding: gzip, deflate, br' \
     -H 'Content-Type: image/jpeg'
```

but I get an `Unsupported Media Type` error:

```bash
HTTP/1.1 415 Unsupported Media Type
Content-Type: application/json; charset=utf-8
Content-Length: 100
Connection: close
Date: Thu, 01 Apr 2021 15:43:45 GMT
vary: Origin
Access-Control-Allow-Origin: *
X-Kong-Upstream-Latency: 1
X-Kong-Proxy-Latency: 709
Via: kong/2.2.1

{"statusCode":"415","error":"Unsupported Media Type","message":"Unsupported Media Type: image/jpeg"}
```

**Comment by michrome** [comment, marked as answer]
Got it 😀

```bash
curl -X POST 'https://xxxxxxxxxx.supabase.co/storage/v1/object/my-bucket1/my-file' \
  -F 'image=@/Users/mike/Desktop/my-local-file.jpg' \
  -H 'Authorization: Bearer xxxxxxxxxx'
```

**Reply by inian** [reply]
Hey Mike, yup! The backend expects FormData. 

Also, if you are directly using the API, we have an autogenerated swagger spec here - `https://yourprojectref.supabase.net/storage/v1/documentation/json`. You can paste the JSON from that endpoint into a swagger viewer and get the documentation like this. 
![image](https://user-images.githubusercontent.com/2155545/113324178-6eb6fa80-9334-11eb-9d11-3086eb8492cd.png)

I am finding a better way to expose this on our website our via a swagger UI endpoint.
